src/gui/sdlmain.cpp

```diff
@@ -3401,51 +3404,56 @@ static SDL_Rect calc_viewport(const int win_w, const int win_h)
 	case IntegerScalingMode::Horizontal: {
 		// Calculate scaling multiplier that will allow to fit the whole
 		// image into the window or viewport bounds.
-		const auto pixel_aspect = round(sdl.draw.height * sdl.draw.scaley) /
-		                          sdl.draw.height;
+		const auto pixel_aspect = round(draw_height * draw_scaley) /
+		                          draw_height;
 		auto integer_scale_factor = std::min(
-		        bounds_w / sdl.draw.width,
-		        static_cast<int>(bounds_h / (sdl.draw.height * pixel_aspect)));
+		        bounds_w / draw_width,
+		        static_cast<int>(bounds_h / (draw_height * pixel_aspect)));
 
 		if (integer_scale_factor < 1) {
 			// Revert to fit to viewport
 			std::tie(view_w, view_h) = calculate_bounded_dims();
 		} else {
 			// Calculate the final viewport.
-			view_w = sdl.draw.width * integer_scale_factor;
-			view_h = iround(sdl.draw.height * integer_scale_factor * pixel_aspect);
+			view_w = draw_width * integer_scale_factor;
+			view_h = iround(draw_height * integer_scale_factor *
+			                pixel_aspect);
```

Comment:
Just trying to understand the nuances of the various roundings (we've got `round()`, `static_cast<int>(...)`, and `iround()` all used here).

For the `static_cast<int>(bounds_h / (draw_height * pixel_aspect)` case, is this where we really need the results like `0.999` to be cast back down to a `0` integer instead of rounding up to `1` (if say, `iround` were used?).

Reply:
In essence, we're dealing with two concepts here:

- rounding numbers, 
- and converting a number to its immediately smaller integer, which is the floor operation.

Both are needed here.

`iround` is just a shortcut so you don't need to cast to an integer type after `round` or `roundf`. I'd say we should introduce a similar `ifloor` helper function as that's a lot more descriptive than `static_cast<int>(...)`.

Now, there's a difference between casting a floating-point number to an integer type vs using the `floor` function:

```c
#include <math.h>
#include <stdio.h>

int main() {
  printf(" (int) 1.6  = %2d\n", (int) 1.6);
  printf(" (int)-1.6  = %2d\n", (int)-1.6);
  printf("floor( 1.6) = %2g\n", floorf( 1.6));
  printf("floor(-1.6) = %2g\n", floor(-1.6));
}
```

Output:

```
% gcc test.c && ./a.out                                                                        [cmd]
 (int) 1.6  =  1
 (int)-1.6  = -1
floor( 1.6) =  1
floor(-1.6) = -2
```

So casting to `int` takes the closest integer closer to zero, while `floor` the closest smaller integer. If we only dealing with positive numbers, this does not matter, which is the case here. 

I will introduce an `iround` helper function that adheres to the `floor` contract.

Reply:
Thanks for the explanation - all makes sense!
That `ifloor` is going to be a big self-documenting improvement about the intent :+1:

Reply:
Done 😎